base/src/Data/Macaw/AbsDomain/AbsState.hs

```diff
@@ -93,8 +92,36 @@ import           Data.Macaw.Types
 ------------------------------------------------------------------------
 -- Utilities
 
-addOff :: NatRepr w -> Integer -> Integer -> Integer
-addOff w o v = toUnsigned w (o + v)
+-- | A helper for computing new signed offsets for values of type 'StackOffsetAbsVal'
+--
+-- This is a bit fiddly because 'StackOffsetAbsVal' has an 'Int64' as its second
+-- value, so we need to be very careful when converting from bitvectors (encoded
+-- as 'Integer') to 'Int64'.
+addSignedOffset
+  :: NatRepr w
```

Comment:
What is the purpose of this `NatRepr w` argument? It's not used anywhere in the implementation. Is the idea that the `Integer` should be a bitvector of length `w`? If so, it might be worth stating this in the Haddocks.